src/lib/sectionList/filters/filterConfig.tsx

```diff
@@ -0,0 +1,51 @@
+import { StringParam } from 'use-query-params'
+import { z } from 'zod'
+import { DataElement } from '../../../types/generated'
+import { IDENTIFIABLE_KEY } from '../../constants'
+import { isValidUid } from '../../models'
+import { CustomDelimitedArrayParam } from './customParams'
+
+const zodArrayIds = z.array(z.string().refine((val) => isValidUid(val)))
+
+/* Zod schema for validation of the decoded params */
+export const filterParamsSchema = z
+    .object({
+        [IDENTIFIABLE_KEY]: z.string(),
+        aggregationType: z.array(z.nativeEnum(DataElement.aggregationType)),
+        domainType: z.array(z.nativeEnum(DataElement.domainType)),
+        valueType: z.array(z.string()),
+        dataSet: zodArrayIds,
+    })
+    .partial()
+
+/* useQueryParams config-map object
+Mapping each filter to a config object that handles encoding/decoding */
+export const filterQueryParamType = {
+    [IDENTIFIABLE_KEY]: StringParam,
+    aggregationType: CustomDelimitedArrayParam,
+    domainType: CustomDelimitedArrayParam,
+    valueType: CustomDelimitedArrayParam,
+    dataSet: CustomDelimitedArrayParam,
+} as const satisfies QueryParamsConfigMap
```

Comment:
I don't love the duplication here. I did experiment generating this dynamically based on the zod-schema. But it turned a bit complex and needed type castings. So this is safer.